disnake/guild.py

```diff
@@ -2165,6 +2166,102 @@ async def create_scheduled_event(
         data = await self._state.http.create_guild_scheduled_event(self.id, reason=reason, **fields)
         return GuildScheduledEvent(state=self._state, data=data)
 
+    async def welcome_screen(self) -> WelcomeScreen:
+        """|coro|
+
+        Retrieves the guild's :class:`WelcomeScreen`.
+
+        Requires the :attr:`~Permissions.manage_guild` permission if the welcome screen is not enabled.
+
+        .. note::
+
+            To determine whether the welcome screen is enabled, check for the
+            presence of ``WELCOME_SCREEN_ENABLED`` in :attr:`Guild.features`.
+
+        .. versionadded:: 2.5
+
+        Raises
+        ------
+        NotFound
+            The welcome screen is not set up, or you do not have permission to view it.
+        HTTPException
+            Retrieving the welcome screen failed.
+
+        Returns
+        -------
+        :class:`WelcomeScreen`
+            The guild's welcome screen.
+        """
+        data = await self._state.http.get_guild_welcome_screen(self.id)
+        return WelcomeScreen(state=self._state, data=data, guild=self)
+
+    async def edit_welcome_screen(
+        self,
+        *,
+        enabled: bool = MISSING,
+        channels: Optional[List[WelcomeScreenChannel]] = MISSING,
+        description: Optional[str] = MISSING,
+        reason: Optional[str] = None,
+    ) -> WelcomeScreen:
+        """|coro|
+
+        This is a lower level method to :meth:`WelcomeScreen.edit` that allows you
+        to edit the welcome screen without fetching it and save an API request.
+
+        All parameters are optional.
```

Comment:
```suggestion
        Requires the guild to be a community guild.
```
same thing for `WelcomeScreen.edit`

Reply:
Actually, it is documented as such.
https://github.com/DisnakeDev/disnake/pull/339/commits/cdd46481b82fcc9f3b68beb87b94495947fe3f3d